### PR TITLE
Add PlatformDropdownButton

### PR DESCRIPTION
**Pull request by bookshiyi**

### CODE REVIEW

File: lib/src/platform_dropdown_button.dart [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                  data?.onChanged(items[index].value);
+                else
+                  onChanged(items[index].value);
+              },
+              children: data?.items != null
+                  ? data!.items.map((e) => e.child).toList()
+                  : items.map((e) => e.child).toList(),
+            ),
+          ),
+        ),
```
Comment (aqwert) [inline lib/src/platform_dropdown_button.dart:200]: A lot of the properties are hard coded and not configurable

File: lib/src/platform_dropdown_button.dart [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      onPressed: () {
+        _showCupertinoPicker();
+      },
+      child: data?.value != null
+          ? Text('${data!.value}')
+          : value != null
+              ? Text('$value')
+              : Text('Cupertino'),
+    );
+  }
```
Comment (aqwert) [inline lib/src/platform_dropdown_button.dart:216]: Issue with this is that it is very opinionated to having used `CupertinoButton`.

**Review by aqwert — COMMENTED** [review]
I have not created a `PlatformDropdownButton` since there is no easy and no unopinionated way of creating a dropdown for cupertino

**Comment by bookshiyi** [comment]
> I have not created a `PlatformDropdownButton` since there is no easy and no unopinionated way of creating a dropdown for cupertino

That's true, But I couldn't think of a better way.

**Pull request closed without merging**